### PR TITLE
adding a "future enhancement" section to the component READMEs

### PR DESCRIPTION
**Pull request by dlockhart**
As discussed in our retro.

We're hoping that calling out the things we know we want to add in the future will encourage folks to contribute those enhancements instead of building them outside of here.

### CODE REVIEW

File: components/list/README.md [hunk 1]
```diff
@@ -92,3 +92,10 @@ The `d2l-list-content` provides additional consistent layout for primary and sec
   </d2l-list-item-content>
 </d2l-list-item>
 ```
+
+## Future Enhancements
+
+- Drag & drop support
+- Paging: integration with "load more", "scroll" and "numeric" paging mechanisms
```
Comment (dlockhart) [inline components/list/README.md]: @m6jones @dbatiste anything else you're aware of that we'd like to add?

File: templates/README.md [hunk 2]
```diff
@@ -41,4 +41,11 @@ Mental Models and Page Structure:
...
 If no nodes are assigned to the `footer` slot, the footer is hidden.
 
-**Note:** this template automatically includes `<header>`, `<main>`, `<aside>` and `<footer>` elements, so there's no need to include them inside the various slots.
\ No newline at end of file
+**Note:** this template automatically includes `<header>`, `<main>`, `<aside>` and `<footer>` elements, so there's no need to include them inside the various slots.
+
+## Future Enhancements
+
+- Optional draggable divider between the primary and secondary panels
+- For smaller devices, the primary and secondary panels will move into collapsible drawers
```
Comment (dlockhart) [inline templates/README.md]: @wongvincent FYI

File: components/list/README.md (hunk 1 above)
Comment (dbatiste) [reply components/list/README.md]: I think Maya is keen to update the list items to use grid for managing focus with arrow keys.

File: components/loading-spinner/README.md [hunk 3]
```diff
@@ -16,3 +16,9 @@
 
 - `color` (String): Color of the animated bar (default is `--d2l-color-celestine`)
 - `size` (Number): Height and width (`px`) of the spinner (default is `50`)
+
+## Future Enhancements
+
+- Ability to "freeze" the spinner at a specified frame to make visual diff testing easier
```
Comment (dbatiste) [inline components/loading-spinner/README.md]: Haha yes!

File: components/button/README.md [hunk 4]
```diff
@@ -86,3 +86,7 @@ Floating workflow buttons behavior can be added by using the `<d2l-floating-butt
 
 - `always-float` (Boolean): Indicates to display buttons as always floating
 - `min-height` (String, default: `'500px'`): The minimum height of the viewport to display floating buttons at (where applicable). If viewport is less than `min-height`, buttons will never appear floating (unless `always-float` is used). If viewport is greater than `min-height` then buttons will float when applicable.
+
+## Future Enhancements
+
```
Comment (dbatiste) [inline components/button/README.md]: Do we want to list preset buttons?

Comment (dlockhart) [reply components/button/README.md]: Hadn't thought about it, but we could... like `<d2l-button-preset type="save">` type thing?

Comment (dbatiste) [reply components/button/README.md]: Yep that's what I was thinking... though I hadn't thought much about the API.

**Pull request merged**